### PR TITLE
Frontend libs 9.3.0

### PR DESCRIPTION
**Pull request by goranalkovic-infinum**
### Added
- 59 new UI icons
- New `GutenbergBlock` helper component, to streamline `-block.js` files
- New `bem` helper to enable less code for selectors that don't need the more advanced conditions

### Updated
- Updated 18 UI icons
- Fixed JSDoc `inlineLabel` prop types within `OptionSelector` and `ColorPallete`
- Fixed `disabled` prop not working within certain variants of `IconToggle`
- Fixed WP 6.4 block names in certain places (e.g. breadcrumbs in Editor)
- Fixed readme for `ResponsiveSlider`
- `WidthOffsetRangeSlider` now has a `colAutoStartOverride` prop
- `WidthOffsetRangeSlider` will behave better when using AUTO offset

### Storybook
- Cleaned up Storybook majorly and migrated it to 7.x

### Important
**Minimum required Node version is now 18!**

<hr />

#### `GutenbergBlock` helper
Allows way simpler block definition files.

Instead of 
```jsx
import React from 'react';
import { MyBlockEditor } from './components/my-block-editor';
import { MyBlockOptions } from './components/my-block-options';
import { InspectorControls } from '@wordpress/block-editor';

export const MyBlock = (props) => (
	return (
		<>
			<InspectorControls>
				<MyBlockOptions {...props} />
			</InspectorControls>
			<MyBlockEditor {...props} />
		</>
	);
};
```
you can now do

```jsx
import React from 'react';
import { MyBlockEditor } from './components/my-block-editor';
import { MyBlockOptions } from './components/my-block-options';
import { GutenbergBlock } from '@eightshift/frontend-libs/scripts';

export const MyBlock = (props) => (
	<GutenbergBlock
		{...props}
		editor={MyBlockEditor}
		options={MyBlockOptions}
	/>
);
```

<hr />

#### `bem` helper

Simplifies `selector` calls where no advanced conditional logic is needed.

| image | `selector(blockClass, blockClass, selectorClass)` |
|---|---|
| image | `bem(blockClass, selectorClass)` |

| image | `selector(blockClass, blockClass, selectorClass, 'lorem')` |
|---|---|
| image | `bem(blockClass, selectorClass, 'lorem')` |

### CODE REVIEW

**Review by iruzevic — APPROVED** [review]

File: .storybook/manager-head.html [hunk 1]
```diff
@@ -68,18 +68,34 @@
...
+		gap: calc(0.25rem + 0.2em) !important;
+	}
+
 	.sidebar-item,
 	.sidebar-subheading {
 		margin-bottom: 1px !important;
 	}
 
-	.sidebar-subheading ~ .sidebar-item {
+	.sidebar-subheading~.sidebar-item {
```
Comment (piqusy) [inline .storybook/manager-head.html:92]: Why the squeeze? Not applied above 🤔
Can it be the same across the file?

File: scripts/components/advanced-color-picker/docs/story.js [hunk 2]
```diff
@@ -1,15 +1,9 @@
...
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
-	title: `Options/AdvancedColorPicker`,
-	parameters: {
-		docs: {
-			page: readme
-		}
-	},
+	title: 'Options/AdvancedColorPicker',
```
Comment (piqusy) [inline scripts/components/advanced-color-picker/docs/story.js:6]:
```suggestion
	title: 'Options/Advanced Color Picker',
```

File: scripts/components/animated-content-visibility/docs/story.js [hunk 3]
```diff
@@ -1,19 +1,12 @@
 import React, { useState } from 'react';
 import { Button } from '@wordpress/components';
 import { AnimatedContentVisibility } from '@eightshift/frontend-libs/scripts';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/AnimatedContentVisibility',
```
Comment (piqusy) [inline scripts/components/animated-content-visibility/docs/story.js:7]:
```suggestion
	title: 'Options/Animated Content Visibility',
```

File: scripts/components/block-inserter/docs/story.js [hunk 4]
```diff
@@ -1,13 +1,7 @@
 import React from 'react';
-import readme from './readme.mdx';
 
 export default {
 	title: 'Options/BlockInserter',
```
Comment (piqusy) [inline scripts/components/block-inserter/docs/story.js:4]:
```suggestion
	title: 'Options/Block Inserter',
```

File: scripts/components/color-palette-custom/docs/story.js [hunk 5]
```diff
@@ -1,16 +1,10 @@
 import React, { useState } from 'react';
-import readme from './readme.mdx';
 import { ColorPalette } from '../color-palette-custom';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 import { icons } from '../../../editor';
 
 export default {
 	title: 'Options/ColorPalette',
```
Comment (piqusy) [inline scripts/components/color-palette-custom/docs/story.js:7]:
```suggestion
	title: 'Options/Color Palette',
```

File: scripts/components/color-picker-component/docs/story.js [hunk 6]
```diff
@@ -1,16 +1,10 @@
 import React from 'react';
-import readme from './readme.mdx';
 import { useState } from '@wordpress/element';
 import { ColorPicker } from '../color-picker-component';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/ColorPicker',
```
Comment (piqusy) [inline scripts/components/color-picker-component/docs/story.js:7]:
```suggestion
	title: 'Options/Color Picker',
```

File: scripts/components/color-swatch/docs/story.js [hunk 7]
```diff
@@ -1,15 +1,9 @@
 import React from 'react';
 import { ColorSwatch } from '@eightshift/frontend-libs/scripts';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/ColorSwatch',
```
Comment (piqusy) [inline scripts/components/color-swatch/docs/story.js:6]:
```suggestion
	title: 'Options/Color Swatch',
```

File: scripts/components/fancy-divider/docs/story.js [hunk 8]
```diff
@@ -1,16 +1,10 @@
...
 import { icons } from '../../../editor';
 
 export default {
-	title: `Options/FancyDivider`,
-	parameters: {
-		docs: {
-			page: readme
-		}
-	},
+	title: 'Options/FancyDivider',
```
Comment (piqusy) [inline scripts/components/fancy-divider/docs/story.js:7]:
```suggestion
	title: 'Options/Fancy Divider',
```

File: scripts/components/help-modal/docs/story.js [hunk 9]
```diff
@@ -1,15 +1,9 @@
 import React from 'react';
-import readme from './readme.mdx';
 import { HelpModal } from '../help-modal';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/HelpModal',
```
Comment (piqusy) [inline scripts/components/help-modal/docs/story.js:6]:
```suggestion
	title: 'Options/Help Modal',
```

File: scripts/components/icon-label/docs/story.js [hunk 10]
```diff
@@ -1,16 +1,10 @@
 import React from 'react';
 import { IconLabel } from '../icon-label';
 import { icons } from '@eightshift/frontend-libs/scripts';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/IconLabel',
```
Comment (piqusy) [inline scripts/components/icon-label/docs/story.js:7]:
```suggestion
	title: 'Options/Icon Label',
```

File: scripts/components/icon-toggle/docs/story.js [hunk 11]
```diff
@@ -2,16 +2,10 @@ import React, { useState } from 'react';
 import { IconToggle } from '../icon-toggle';
 import { icons } from '@eightshift/frontend-libs/scripts';
 import { __ } from '@wordpress/i18n';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/IconToggle',
```
Comment (piqusy) [inline scripts/components/icon-toggle/docs/story.js:8]:
```suggestion
	title: 'Options/Icon Toggle',
```

File: scripts/components/icon-toggle/docs/story.js [hunk 12]
```diff
@@ -24,8 +18,8 @@ export const component = () => {
 			<div className='es-display-flex es-flex-wrap es-gap-5!'>
 				<SingleItemShowcase title='Basic toggle'>
 					<IconToggle
-						icon={icons.width}
-						label='Full width'
+						icon={icons.fullMaxShield}
+						label='Full max'
```
Comment (piqusy) [inline scripts/components/icon-toggle/docs/story.js:22]: Full max reloaded 💪🏼

File: scripts/components/link-edit-component/docs/story.js [hunk 13]
```diff
@@ -1,16 +1,10 @@
 import React from 'react';
-import readme from './readme.mdx';
 import { useState } from '@wordpress/element';
 import { LinkEditComponent } from '../link-edit-component';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/LinkEditComponent',
```
Comment (piqusy) [inline scripts/components/link-edit-component/docs/story.js:7]:
```suggestion
	title: 'Options/Link Edit Component',
```

File: scripts/components/matrix-align-control/docs/story.js [hunk 14]
```diff
@@ -1,18 +1,11 @@
 import React, { useState } from 'react';
-import readme from './readme.mdx';
 import { MatrixAlignControl } from '../matrix-align-control';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/MatrixAlignControl',
```
Comment (piqusy) [inline scripts/components/matrix-align-control/docs/story.js:6]:
```suggestion
	title: 'Options/Matrix Align Control',
```

File: scripts/components/number-picker/docs/story.js [hunk 15]
```diff
@@ -1,16 +1,10 @@
 import React, {useState} from 'react';
-import readme from './readme.mdx';
 import { NumberPicker } from '../number-picker';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 import { icons } from '../../../editor';
 
 export default {
 	title: 'Options/NumberPicker',
```
Comment (piqusy) [inline scripts/components/number-picker/docs/story.js:7]:
```suggestion
	title: 'Options/Number Picker',
```

File: scripts/components/option-selector/docs/story.js [hunk 16]
```diff
@@ -1,15 +1,9 @@
 import React, { useState } from 'react';
 import { icons, ucfirst } from '@eightshift/frontend-libs/scripts';
-import readme from './readme.mdx';
 import { OptionSelector } from '../option-selector';
 
 export default {
 	title: 'Options/OptionSelector',
```
Comment (piqusy) [inline scripts/components/option-selector/docs/story.js:6]:
```suggestion
	title: 'Options/Option Selector',
```

File: scripts/components/popover-with-trigger/docs/story.js [hunk 17]
```diff
@@ -1,16 +1,10 @@
 import React from 'react';
 import { Button } from '@wordpress/components';
 import { PopoverWithTrigger } from '@eightshift/frontend-libs/scripts';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/PopoverWithTrigger',
```
Comment (piqusy) [inline scripts/components/popover-with-trigger/docs/story.js:7]:
```suggestion
	title: 'Options/Popover With Trigger',
```

File: scripts/components/preset-picker/docs/story.js [hunk 18]
```diff
@@ -1,16 +1,10 @@
 import React, { useState } from 'react';
 import { PresetPicker } from '../preset-picker';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 import { icons } from '../../../editor';
 
 export default {
 	title: 'Options/PresetPicker',
```
Comment (piqusy) [inline scripts/components/preset-picker/docs/story.js:7]:
```suggestion
	title: 'Options/Preset Picker',
```

File: scripts/components/responsive-number-picker/docs/story.js [hunk 19]
```diff
@@ -1,16 +1,10 @@
 import React, { useState } from 'react';
 import { ResponsiveNumberPicker } from '../responsive-number-picker';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 import { icons } from '../../../editor';
 
 export default {
 	title: 'Options/ResponsiveNumberPicker',
```
Comment (piqusy) [inline scripts/components/responsive-number-picker/docs/story.js:7]:
```suggestion
	title: 'Options/Responsive Number Picker',
```

File: scripts/components/responsive-slider/docs/story.js [hunk 20]
```diff
@@ -1,16 +1,10 @@
 import React, { useState } from 'react';
 import { ResponsiveSlider } from '../responsive-slider';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 import { icons } from '../../../editor';
 
 export default {
 	title: 'Options/ResponsiveSlider',
```
Comment (piqusy) [inline scripts/components/responsive-slider/docs/story.js:7]:
```suggestion
	title: 'Options/Responsive Slider',
```

File: scripts/components/responsive-toggle-button/docs/story.js [hunk 21]
```diff
@@ -1,16 +1,10 @@
 import React, { useState } from 'react';
 import { ResponsiveToggleButton } from '../responsive-toggle-button';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 import { icons } from '../../../editor';
 
 export default {
 	title: 'Options/ResponsiveToggleButton',
```
Comment (piqusy) [inline scripts/components/responsive-toggle-button/docs/story.js:7]:
```suggestion
	title: 'Options/Responsive Toggle Button',
```

File: scripts/components/server-side-render/docs/story.js [hunk 22]
```diff
@@ -1,14 +1,8 @@
 import React from 'react';
-import readme from './readme.mdx';
 import { ServerSideRender } from '../server-side-render';
 
 export default {
 	title: 'Options/ServerSideRender',
```
Comment (piqusy) [inline scripts/components/server-side-render/docs/story.js:5]:
```suggestion
	title: 'Options/Server Side Render',
```

File: scripts/components/tile-button/docs/story.js [hunk 23]
```diff
@@ -1,16 +1,10 @@
 import React, { useState } from 'react';
 import { __ } from '@wordpress/i18n';
 import { icons, TileButton } from '@eightshift/frontend-libs/scripts';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/TileButton',
```
Comment (piqusy) [inline scripts/components/tile-button/docs/story.js:7]:
```suggestion
	title: 'Options/Tile Button',
```

File: scripts/components/toolbar-option-picker/docs/story.js [hunk 24]
```diff
@@ -1,14 +1,8 @@
 import React from 'react';
-import readme from './readme.mdx';
 import { ToolbarOptionPicker } from '../toolbar-option-picker';
 
 export default {
 	title: 'Options/ToolbarOptionPicker',
```
Comment (piqusy) [inline scripts/components/toolbar-option-picker/docs/story.js:5]:
```suggestion
	title: 'Options/Toolbar Option Picker',
```

File: scripts/components/use-toggle/docs/story.js [hunk 25]
```diff
@@ -1,14 +1,8 @@
 import React, { useState } from 'react';
 import { UseToggle } from '../use-toggle';
-import readme from './readme.mdx';
 
 export default {
 	title: 'Options/UseToggle',
```
Comment (piqusy) [inline scripts/components/use-toggle/docs/story.js:5]:
```suggestion
	title: 'Options/Use Toggle',
```

File: scripts/components/width-offset-range-slider/docs/story.js [hunk 26]
```diff
@@ -1,15 +1,9 @@
 import React, { useState } from 'react';
 import { WidthOffsetRangeSlider } from '@eightshift/frontend-libs/scripts';
-import readme from './readme.mdx';
 import { SingleItemShowcase } from '../../../storybook/helpers';
 
 export default {
 	title: 'Options/WidthOffsetRangeSlider',
```
Comment (piqusy) [inline scripts/components/width-offset-range-slider/docs/story.js:6]:
```suggestion
	title: 'Options/Width Offset Range Slider',
```

**Review by piqusy — DISMISSED** [review]
Nice work 👏🏼 

Left a "few" suggestions 😄 Mostly the same for storybook TItle

File: .storybook/manager-head.html (hunk 1 above)
Comment (goranalkovic-infinum) [reply .storybook/manager-head.html:92]: Damnit, auto-formatting 😅 
I'll fix it

File: scripts/components/advanced-color-picker/docs/story.js (hunk 2 above)
Comment (goranalkovic-infinum) [reply scripts/components/advanced-color-picker/docs/story.js:6]: I wanted to keep these as-is, so they match the actual component name

**Review by iobrado — APPROVED** [review]

**Review by piqusy — APPROVED** [review]

**Review by iobrado — APPROVED** [review]

**Pull request merged**